content/en/docs/setup/install/operator/index.md

```diff
@@ -123,6 +123,12 @@ Now, with the controller running, you can change the Istio configuration by edit
 the `IstioOperator` resource. The controller will detect the change and respond by updating
 the Istio installation correspondingly.
 
+{{< warning >}}
+The current behavior of the Istio operator is to _merge_ configuration of some components rather than to _override_
+configuration. This applies to components that have immutable fields such as the Kubernetes `Service` object's
+`clusterIP` field. This can cause Istio components to display a mix of current configuration and previous configuration.
```

Comment:
Yes, I think this could use a more specific example. What kind of "merge" happens when you try to update a field of a Service object? Is it only the immutable fields that have this issue? What exactly happens if you change the clusterIP of a service field, how does it merge 2 clusterIP valuse?

Probably should also mention a workaround and/or ETA for a fix??
cc @ostromart

Reply:
it's a bit unclear to me what the difference between merge and override is, since merge is an override. an example would be good explaining that distinction.

Reply:
@frankbu @ostromart not the best example, I will remove it. This was the example given to me in the original issue I raised, but it isn't helpful here.

For reference, here is the original issue, the comments add context: https://github.com/istio/istio/issues/27032

Is the example I provided there (modifying pod anti-affinity settings, and ending up with a merged result) a more appropriate example?

As far as a workaround, there really isn't any, which is why I wanted to add the warning. As you can see in the linked issue above, the suggested workaround is to delete the `IstioOperator` resource and create a new one, which clearly isn't feasible if you want your Istio installation to stay up.

Happy to include an ETA for the fix, it's basically whenever Istio drops support for K8s 1.16 (k8s 1.17 and higher have server side apply, which solves this issue), which if I'm not mistaken will be Istio 1.9?

Reply:
Actually, it's possible that this has been resolved. Waiting on info in istio/istio#27032